demos/keyboard-nav/index.html

```diff
@@ -0,0 +1,224 @@
+      <variable type="" id="8]s[S+Gy+%k7HoFup])m">item</variable>
+    </variables>
+    <block type="controls_if" id="^^hRiX{cCPHWJr}?8M^5" x="37" y="162">
+      <value name="IF0">
+        <block type="logic_compare" id="#yzPl3=N6CIpUKvJ0b1(">
+          <field name="OP">EQ</field>
+          <value name="A">
+            <block type="math_arithmetic" id="h#NOs2/B*.f{5N=9L|m|">
+              <field name="OP">ADD</field>
+              <value name="A">
+                <shadow type="math_number" id="~4Z2|fDhdniR@Z=9m.]5">
+                  <field name="NUM">1</field>
+                </shadow>
+              </value>
+              <value name="B">
+                <shadow type="math_number" id="X9ywC|L-2$E#DHd[75bJ">
+                  <field name="NUM">1</field>
+                </shadow>
+              </value>
+            </block>
+          </value>
+          <value name="B">
+            <block type="math_single" id="Ad|PE{k|OX2k,ef#S1sC">
+              <field name="OP">ROOT</field>
+              <value name="NUM">
+                <shadow type="math_number" id=",d-)d1gr[f%EYC)jryuz">
+                  <field name="NUM">9</field>
+                </shadow>
+                <block type="math_number" id="xn[NF$ps1cPK(;q8K1PM">
+                  <field name="NUM">123</field>
+                </block>
+              </value>
+            </block>
+          </value>
+        </block>
+      </value>
+      <statement name="DO0">
+        <block type="lists_setIndex" id="EX~e8#P8]I7BOSH%]11H">
+          <mutation at="true"></mutation>
+          <field name="MODE">SET</field>
+          <field name="WHERE">FROM_START</field>
+          <value name="LIST">
+            <block type="variables_get" id="#(o=|qGel$T_MJkZU,%1">
+              <field name="VAR" id="~GNXm@Z(wclI]t3zTf.g" variabletype="">list</field>
+            </block>
+          </value>
+          <next>
+            <block type="text_append" id="gB,Xu47i^C//_e-J7NiY">
+              <field name="VAR" id="8]s[S+Gy+%k7HoFup])m" variabletype="">item</field>
+              <value name="TEXT">
+                <shadow type="text" id="|PcJ2q-,B,gRn7Bz%gm|">
+                  <field name="TEXT"></field>
+                </shadow>
+              </value>
+            </block>
+          </next>
+        </block>
+      </statement>
+      <next>
+        <block type="controls_repeat_ext" id="O=qqWbd:drvugOFlDR3R">
+          <value name="TIMES">
+            <shadow type="math_number" id="0Y^xJhJqlh$.1C?5}B`;">
+              <field name="NUM">10</field>
+            </shadow>
+          </value>
+        </block>
+      </next>
+    </block>
+  </xml>
+
+  <script>
+    var demoWorkspace = Blockly.inject('blocklyDiv',
+        {media: '../../media/',
+         toolbox: document.getElementById('toolbox')});
+    Blockly.Xml.domToWorkspace(document.getElementById('startBlocks'),
+                               demoWorkspace);
+
+    function validNode(node) {
```

Comment:
Document functions with explanations of what each function does. Demos are better if they include extra documentation. ;)